src/tools/r2rdump/CoreDisTools.cs

```diff
@@ -76,4 +76,91 @@ public static IntPtr GetDisasm(Machine machine)
             return InitBufferedDisasm(target);
         }
     }
+
+    public class Disassembler : IDisposable
+    {
+        private readonly IntPtr _disasm;
+
+        private readonly byte[] _image;
+
+        private readonly Machine _machine;
+
+        public Disassembler(byte[] image, Machine machine)
+        {
+            _disasm = CoreDisTools.GetDisasm(machine);
+            _image = image;
+            _machine = machine;
+        }
+
+        public void Dispose()
+        {
+            if (_disasm != IntPtr.Zero)
+            {
+                CoreDisTools.FinishDisasm(_disasm);
+            }
+        }
+
+        public int GetInstruction(RuntimeFunction rtf, int imageOffset, int rtfOffset, out string instruction)
+        {
+            int instrSize = CoreDisTools.GetInstruction(_disasm, rtf, imageOffset, rtfOffset, _image, out instruction);
+
+            switch (_machine)
+            {
+                case Machine.Amd64:
+                case Machine.IA64:
```

Comment:
The GetDisasm() function returns a CorDisasm object based on the TargetArch. Since IA64 is not a target arch recognized by coredistools, the CorDisasm object might not be for the right architecture. Would that be a problem?

Reply:
Well, if we don't find the right disassembler for the current Machine, we're most likely done one way or another, I don't think my change modifies anything in this respect. The only thing I added is a switch to post-process the disassembly output in some specific cases.